.vitepress/theme/components/VueMasteryModal.vue

```diff
@@ -0,0 +1,195 @@
+<script setup lang="ts">
+import { watch } from 'vue'
+
+const VIDEO_SOURCE = 'https://player.vimeo.com/video/647441538?autoplay=1'
+let showWhyVue: boolean = $ref(false)
+
+watch(
+  () => showWhyVue,
+  (value) => {
+    value
+      ? (document.documentElement.style.overflow = 'hidden')
+      : (document.documentElement.style.overflow = 'auto')
+  }
+)
+</script>
+
+<template>
+  <a
+    id="vuemastery-action"
+    class="why-vue"
+    @click="showWhyVue = true"
+    href="javascript:;"
+  >
+    <svg
+      class="icon-play"
+      aria-labelledby="simpleicons-play-icon"
+      role="img"
+      viewBox="0 0 100 125"
+      fill="#FFFFFF"
+    >
+      <title id="simpleicons-play-icon" lang="en">Play icon</title>
+      <path
+        d="M50,3.8C24.5,3.8,3.8,24.5,3.8,50S24.5,96.2,50,96.2S96.2,75.5,96.2,50S75.5,3.8,50,3.8z M71.2,53.3l-30.8,18  c-0.6,0.4-1.3,0.5-1.9,0.5c-0.6,0-1.3-0.1-1.9-0.5c-1.2-0.6-1.9-1.9-1.9-3.3V32c0-1.4,0.8-2.7,1.9-3.3c1.2-0.6,2.7-0.6,3.8,0  l30.8,18c1.2,0.6,1.9,1.9,1.9,3.3S72.3,52.7,71.2,53.3z"
+      ></path>
+    </svg>
+    Why Vue
+  </a>
+  <Teleport to="body">
```

Comment:
The use of Teleport is currently causing a hydration mismatch in production build. I think it can be fixed if the `v-if` is placed on the Teleport itself.